pkg/block/index.go

```diff
@@ -347,9 +407,50 @@ func Repair(dir string, id ulid.ULID) (resid ulid.ULID, err error) {
 
 var castagnoli = crc32.MakeTable(crc32.Castagnoli)
 
+func IgnoreCompleteOutsideChunk(mint, maxt int64, _ *chunks.Meta, curr *chunks.Meta) (bool, error) {
+	if curr.MinTime > maxt || curr.MaxTime < mint {
+		// "Complete" outsider. Ignore.
+		return true, nil
+	}
+	return false, nil
+}
+
+func IgnoreIssue347OutsideChunk(mint, maxt int64, _ *chunks.Meta, curr *chunks.Meta) (bool, error) {
+	if curr.MinTime == maxt && curr.MaxTime > maxt {
+		// "Near" outsider from issue https://github.com/prometheus/tsdb/issues/347. Ignore.
+		return true, nil
+	}
+	return false, nil
+}
+
+func IgnoreDuplicateOutsideChunk(mint, maxt int64, prev *chunks.Meta, curr *chunks.Meta) (bool, error) {
+	if prev == nil {
+		return false, nil
+	}
+
+	if curr.MinTime > prev.MaxTime {
```

Comment:
Can you have `curr.Mintime = prev.MaxTime`?

Reply:
no. Chunks never overlaps. Blocks overlaps with miliesecond always (TSDB known inconsistency). But good question.